pkg/reconciler/subscription/subscription.go

```diff
@@ -196,8 +196,8 @@ func (r *Reconciler) resolveSubscriber(ctx context.Context, subscription *v1.Sub
 	// Resolve Subscriber.
 	subscriber := subscription.Spec.Subscriber.DeepCopy()
 	if !isNilOrEmptyDestination(subscriber) {
-		// Populate the namespace for the subscriber since it is in the namespace
-		if subscriber.Ref != nil {
+		// Populate the namespace for the subscriber if required
+		if subscriber.Ref != nil && subscriber.Ref.Namespace == "" {
 			subscriber.Ref.Namespace = subscription.Namespace
```

Comment:
I see that the defaulting was always happening here but it may be better to move it to the defaulting [webhook](https://github.com/knative/eventing/blob/main/pkg/apis/messaging/v1/subscription_defaults.go) for subscriptions.

Reply:
Thanks for the review. I agree the defaulting webhook would be a better place. Let me move it...

Reply:
This defaulting should be kept for backward compatibility since existing resources won't have it defaulted until an update happens

Reply:
@pierDipi: So we don't we resync the resources after a couple of time?

Reply:
resync is not necessarily producing an update (knative/pkg tries to reduce updates as much as possible to reduce API server load) therefore the webhook is not always applied

Reply:
This is only a `status` update: https://github.com/knative/eventing/blob/d4184192c6a7ff995f27ebd7b74c6d6e3d470fd2/pkg/client/injection/reconciler/messaging/v1/subscription/reconciler.go#L311-L341, so spec changes in `ReconcileKind` are not persisted.

In addition, on upgrade controller and webhook are not atomically updated at the same time, so during this time period we might not have this logic running anywhere